pkg/test/expectations.go

```diff
@@ -0,0 +1,56 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"encoding/base64"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/gomega" // nolint:revive,stylecheck
+)
+
+func (env Environment) ExpectLaunchTemplatesCreatedWithUserDataContaining(substrings ...string) {
+	Expect(env.EC2API.CalledWithCreateLaunchTemplateInput.Len()).To(BeNumerically(">=", 1))
```

Comment:
I'm wondering if we should couple this method to the environment as opposed to the API directly. Thoughts? Like a method on the EC2API that validates all launch templates? Or even maybe just a method to range over the input like `ForEach`

Reply:
hmm I don't have a strong opinion. I slightly favor the way I have it here since it matches the e2e test setup. There are differences in why it's setup like that in the e2e tests, but I kind of like the decoupling of an expectations helper on the env rather than coupling directly with the mocks.

Reply:
I'm not sure that setting the expectation on the mocks makes sense but I think an iteration mechanism on the input as a general concept makes a lot of sense. WDYT?